### PR TITLE
Docs: Add documentation about the frontend sandbox

### PR DESCRIPTION
**Pull request by academo**
**What is this feature?**

Adds documentation about the plugins frontend sandbox

### CODE REVIEW

**Review by irenerl24 — APPROVED** [review]

File: docs/sources/administration/plugin-management/index.md [hunk 1]
```diff
@@ -228,6 +228,57 @@ WARN[06-01|16:45:59] Running an unsigned plugin   pluginID=<plugin id>
...
+{{% /admonition %}}
+
+### Compatibility
+
+The Frontend Sandbox is compatible with all types of plugins including app plugins, panel plugins, and data source plugins.
+
+The Frontend Sandbox is not compatible with:
+
+- Angular-based plugins
+- Plugins developed and signed by Grafana Labs
```
Comment (andresmgot) [inline docs/sources/administration/plugin-management/index.md:262]: WDYT about this suggestion? I found it a bit weird to say that Grafana plugins are not compatible, while may be, it's just a opinionated decission.

```suggestion
The Frontend Sandbox is compatible with all types of plugins including app plugins, panel plugins, and data source plugins. Angular-based plugins are not supported. Plugins developed and signed by Grafana Labs are excluded and cannot be sandboxed.
```

File: docs/sources/administration/plugin-management/index.md [hunk 2]
```diff
@@ -228,6 +228,57 @@ WARN[06-01|16:45:59] Running an unsigned plugin   pluginID=<plugin id>
...
+
+### When to Use Frontend Sandbox
+
+We strongly recommend enabling the Frontend Sandbox for plugins that allow users to write custom JavaScript code for data visualization or manipulation. These plugins, while powerful, can potentially execute arbitrary JavaScript code in your Grafana instance. The sandbox provides an additional layer of security by restricting what this code can access and modify.
+
+Examples of plugins where the sandbox is particularly important include:
+
+- Panel plugins that allow users to write custom JavaScript code
+- Plugins from third parties (other than Grafana Labs)
```
Comment (andresmgot) [inline docs/sources/administration/plugin-management/index.md:271]:
```suggestion
- Plugins from untrusted sources
```

**Review by andresmgot — APPROVED** [review]
LGTM, just a couple of suggestions.

File: docs/sources/administration/plugin-management/index.md (hunk 1 above)
Comment (academo) [reply docs/sources/administration/plugin-management/index.md:262]: I like it!! added

**Pull request merged**